### DISCUSSION TITLE
Is is possible to add a study/indicator to the chart?

### DISCUSSION THREAD

**Discussion by timocov** [tradingview/lightweight-charts#647] [Q&A] [answered]
How to add a study/indicator to the chart?

**Comment by timocov** [comment, marked as answer]
This is related to https://github.com/tradingview/lightweight-charts/discussions/633.

Yes, it is possible.

`lightweight-charts` knows nothing about what exactly you display in the chart - for the library all data sources are series (with different types of the data/displaying).

If you'd like to add several indicators to the chart, just create several series (one series for one single-valued study) with their own options and study/indicator's data.

If your study has several outputs (like Bollinger Bands), you can create several series - one series for one study output/plot. But in this case you need to pay attention on what price scale that series are using - quite probably you want to be sure that all a study's outputs are attached to the same price scale. To do it, use [`priceScaleId`](https://github.com/tradingview/lightweight-charts/blob/master/docs/series-basics.md#parameters) (all outputs should have the same price scale id in this case, regardless it is `'right'`/`'left'` or a random overlay price scale id).

**Reply by big-javascript-fan** [reply]
Thank you so much. timocov.
May I get the sample code or any example?
Thanks

**Reply by timocov** [reply]
@big-javascript-fan I think that the simplest study is Volume study and you can check it here https://jsfiddle.net/TradingView/cnbamtuh/. I believe that all other studies could be done in similar way - you just need to process your data via a study function (you can use any available in npm library if you'd like) and then provide the result of this data to lightweight-charts. For example https://codesandbox.io/s/brave-silence-sg5h9?file=/src/index.js.

**Comment by aradhya-tiwari** [comment]
https://www.youtube.com/watch?v=NlHjhmIe1EI 

https://jsfiddle.net/adrianntf/6qea5ytv/

**Comment by dokocat** [comment]
Is there a way we can get access to the regular indicators provided my tradingview pro charts? Would love to be able to offer users a dropdown they could select different indicators.

**Reply by romfrancois** [reply]
Lightweight Charts doesn't have the same concept of indicators as the other library. 
With Lightweight Charts you have to implement any indicator you'd like to offer/use. Most if not all formula can be obtain online on websites such as Investopedia. Given you're handling already the data you should be able to implement what you want.

**Reply by dokocat** [reply]
Thanks for the info sir! In the upcoming plugins system would it be possible to implement community made indicators?

**Reply by romfrancois** [reply]
Yes most likely some indicators could/will be provided as plugins while some others might be developed by some other devs.
